src/Confluent.Kafka/Consumer.cs

```diff
@@ -855,5 +855,23 @@ internal Consumer(ConsumerBuilder<TKey, TValue> builder)
         /// <inheritdoc/>
         public ConsumeResult<TKey, TValue> Consume(TimeSpan timeout)
             => Consume(timeout.TotalMillisecondsAsInt());
+
+
+        /// <inheritdoc/>
+        public IConsumerGroupMetadata ConsumerGroupMetadata
+        {
+            get
+            {
+                var cgmd = this.kafkaHandle.GetConsumerGroupMetadata();
```

Comment:
Do note that consumer_group_metadata() may return NULL if called on a producer instance or there is no group.id configured.

Reply:
neither of these cases possible - `rd_kafka_poll_set_consumer`, which is called in the consumer constructor, forces that `group.id` must be set, and there is no way for a user to call this method on a producer `rk`.

note that in the docs for `IConsumer.ConsumerGroupMetadata` I did document that this will be `null` if `GroupId` is not set, however I did that to ensure docs wouldn't go out of date in the event this got fixed (i considered the miss-documentation harmless, as obvious exceptions will be thrown on consumer startup in this case not causing confusion).